src/efi.rs

```diff
@@ -265,31 +258,48 @@ impl Component for Efi {
     /// Given an adoptable system and an update, perform the update.
     fn adopt_update(
         &self,
-        sysroot: &openat::Dir,
+        sysroot: &RootContext,
         updatemeta: &ContentMetadata,
     ) -> Result<InstalledContent> {
-        let Some(meta) = self.query_adopt()? else {
+        let esp_devices = blockdev::find_colocated_esps(&sysroot.devices)?;
+        let Some(meta) = self.query_adopt(&esp_devices)? else {
             anyhow::bail!("Failed to find adoptable system")
         };
 
-        let esp = self.open_esp()?;
-        validate_esp(&esp)?;
+        // Confirm that esp_devices is Some(value)
+        let esp_devices = esp_devices.unwrap();
```

Comment:
But we also confirm that just below with a runtime error and not a panic, which I think is right; so we can just drop this line.

Reply:
Can not drop as will get error, use `.unwrap_or_default()` instead.